sim/dex-data.js

```diff
@@ -110,8 +110,51 @@ class Effect {
 		 * @type {number}
 		 */
 		this.gen = 0;
+		/**
+		 * The duration of the effect.
+		 * @type {?number}
+		 */
+		this.duration = this.duration;
+		/**
+		 * Whether or not the effect is ignored by Baton Pass.
+		 * @type {boolean}
+		 */
+		this.noCopy = this.noCopy;
+		/**
+		 * The status that the effect may cause.
+		 * @type {string}
+		 */
+		this.status = this.status;
```

Comment:
This (as well as `weather`, `drain`, `flags` etc) should all be attached to `Move`, not `Effect`). Is there code assuming effects are necessarily moves? If so, leave this and I'll fix it myself at some later point.

Reply:
<code>flags</code> is just being moved here from https://github.com/Zarel/Pokemon-Showdown/blob/master/sim/battle.js#L1605, which prevents crashes in <code>onAfterDamage</code>. For the rest though, yeah there is some code that makes that assumption.

Reply:
Remind me to fix that later.